src/sumlog.jl

```diff
@@ -1,42 +1,94 @@
 """
-$(SIGNATURES)
+    sumlog(X::AbstractArray{T}; dims)
 
-Compute `sum(log.(X))` with a single `log` evaluation.
+Compute `sum(log.(X))` with a single `log` evaluation,
+provided `float(T) <: AbstractFloat`.
 
-This is faster than computing `sum(log.(X))` or even `sum(log, X)`, in
-particular as the size of `X` increases.
-
-This works by representing the `j`th element of `X` as ``x_j = a_j  2^{b_j}``,
+This is faster than computing `sum(log, X)`, especially for large `X`.
+It works by representing the `j`th element of `X` as ``x_j = a_j  2^{b_j}``,
 allowing us to write
 ```math
 \\sum_j \\log{x_j} = \\log(\\prod_j a_j) + \\log{2} \\sum_j b_j
 ```
-Since ``\\log{2}`` is constant, `sumlog` only requires a single `log`
-evaluation.
 """
-function sumlog(x)
-    T = float(eltype(x))
-    _sumlog(T, values(x))
+sumlog(x::AbstractArray{T}; dims=:) where T = _sumlog(float(T), dims, x)
```

Comment:
I think it would be better to keep this restricted to `Real` as other methods in LogExpFunctions?

In any case, the type parameter is not necessary it seems and the dispatch on `dims` is not needed (see below).
```suggestion
sumlog(x::AbstractArray{<:Real}; dims=:) = _sumlog(float(eltype(x)), x; dims=dims)
```

Reply:
Real would be fine. The tests use complex numbers only as examples of things for which `float(x)` isn't an AbstractFloat. (Which in reality could be weird numbers like Dual, etc.)